### DISCUSSION TITLE
Are there available metrics regarding the redelivered and deleted messages?

### DISCUSSION THREAD

**Discussion by albionb96** [nats-io/nats-server#5440] [Q&A]
Hi there,

I wanted to ask if there are already available metrics for the the redelivered and deleted messages, especially when installing nats with helm charts and prom-exporter?

I mean I can see such values when using the nats-box like below:
![image](https://github.com/nats-io/nats.java/assets/49485798/90250b68-c9b3-4b93-a7d5-9dc40a3b43b6)
![image](https://github.com/nats-io/nats.java/assets/49485798/f6f6c169-9011-4fd4-9fb5-6c892b0a2a03)

but I am not sure if there are prom metrics available for this data.

One metric that I was available to find out is called: nats_consumer_num_redelivered, and maybe this covers the redelivered part but regarding the other metrics nothing came up.

I once asked regarding some of these metrics, and it was mentioned that somebody is going to look at it but I don't know if there are any new developments: https://github.com/nats-io/nats-server/discussions/5096#discussioncomment-8592153

**Comment by Jarema** [comment]
Hey!

We will take a look at current exposed metrics and update any missing ones.
I believe those you mentioned should be doable withing metrics.
